### DISCUSSION TITLE
Plot Panel: Add possibility to display static polygon on top of plot

### DISCUSSION THREAD

**Discussion by tonynajjar** [foxglove/community#803] [Q&A]
As the title says this is a feature request for the Plot panel to enable displaying a static definable polygon overlayed on the plot. Ideally, we would input the message path to that polygon (e.g. geometry_msgs/msg/Polygon).

My specific use case: using the Plot panel, I plot speed vs steering angle of my robot. The combination of these two values should not go outside of a certain "envelope" (represented by a polygon). I'd like to visualize this envelope/polygon on the plot to catch when the combination of speed and steering angle goes outside of it.

**Comment by tonynajjar** [comment]
I guess this is similar to https://github.com/orgs/foxglove/discussions/176 and https://github.com/orgs/foxglove/discussions/471

**Comment by defunctzombie** [comment]
While we don't have the ability to specify this value via a message - you can enter a fixed value in the "message path" for a plot series and that will display a dashed line:

Here's an example with some sample IMU data where I set 0.8 as the "Max" series and it shows a line.

<img width="1511" alt="image" src="https://github.com/foxglove/community/assets/84792/a01b163a-1260-4f52-8a85-7b8af22f7eab">

**Comment by tonynajjar** [comment]
Thanks for the tip but I don't think that would do it for me, I need something like this:

![image](https://github.com/foxglove/community/assets/14879660/685dbf26-74c9-4849-acbb-7f934f1c0e48)

I edited in the red square, in reality the polygon is more complex than that. I think your [issue](https://github.com/orgs/foxglove/discussions/176) would be the best way to achieve this

**Comment by indirectlylit** [comment]
This is now feasible using a recently-shipped feature which allows you to set X values independently for each series in path-based (accumulated and current) XY plots. It's available on web now, and will be in the next desktop release.

Here is a (somewhat contrived) example from our internal test data where we draw a rectangle on top of other data using a series of four messages:

> <img width="1340" alt="image" src="https://github.com/user-attachments/assets/d5b94752-7d20-4f18-8613-05bd07cff3e4" />


I don't think this is necessarily the ideal solution to adding annotations to a plot, but it's a bit of an improvement.
